src/host/test/ledger.cpp

```diff
@@ -1,61 +1,685 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
These unit tests should test all edge cases that our end-to-end tests do not. 

They use the `TestEntrySubmitter` class to write to/truncate the ledger. This class allows us to trivially verify that something was written to the ledger the value of the transaction index is written in the ledger itself. On read, it is then easy to verify that when asking for all entries between `from` and `to` that the values retrieves are `from`, `from+1`, `from+2`, ..., `to`.